actors/miner/tests/util.rs

```diff
@@ -1781,6 +1782,9 @@ impl ActorHarness {
             penalty_from_vesting -= cfg.penalty_from_unlocked.clone();
 
             pledge_delta -= penalty_from_vesting;
+            // if we have no unvested rewards, we can't unlock them so there's no locked/pledge delta,
+            // we take what we can from unvested and then the rest remains as fee_debt
+            pledge_delta = std::cmp::max(pledge_delta, -unvested_vesting_funds(rt, &state));
```

Comment:
I think this is definitionally violating the purpose of cfg.penalty_from_unlocked.  That field is not like the others which are tallying up what the penalty will be, it is asserting that this much penalty comes from unlocked as behavior.  If we massage the test code to not assert this anymore we need to remove that part of the config.  It always seemed overspecified to me anyway so that's probably an improvement.

Reply:
yeah, I agree, there's already awkwardness in here and it felt like I was embedding too much of the protocol logic in the harness itself but didn't want to undertake a deeper refactor!

I'll give it a go through but might have to timebox the effort.